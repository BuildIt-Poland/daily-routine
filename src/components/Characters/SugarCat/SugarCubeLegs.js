import React from 'react';
import styled, { keyframes } from 'styled-components';

import SugarCubeLeftLegArtwork from './SugarCubeLeftLegArtwork';
import SugarCubeRightLegArtwork from './SugarCubeRightLegArtwork';

const leftLegWaveAnimation = keyframes`
  0% { transform: translateX(0) rotate(-5deg); }
  50% { transform: translateX(-8px) rotate(35deg); }
  100% { transform: translateX(0) rotate(-5deg); }
`;

const rightLegWaveAnimation = keyframes`
  0% { transform: translateX(0) rotate(-5deg); }
  50% { transform: translateX(-9px) rotate(25deg); }
  100% { transform: translateX(0) rotate(-5deg); }
`;

const SugarCubeLeftLeg = styled(SugarCubeLeftLegArtwork)`
  position: absolute;
  bottom: 2.1rem;
  left: 23rem;
  animation: ${rightLegWaveAnimation} 2.5s ease-in-out -0.3s infinite;
`;

const SugarCubeRightLeg = styled(SugarCubeRightLegArtwork)`
  position: absolute;
  bottom: 2rem;
  left: 23.8rem;
  animation: ${leftLegWaveAnimation} 2.5s ease-in-out infinite;
`;

function SugarCubeLegs() {
  return (
    <>
      <SugarCubeLeftLeg />
      <SugarCubeRightLeg />
    </>
  );
}

export default SugarCubeLegs;
