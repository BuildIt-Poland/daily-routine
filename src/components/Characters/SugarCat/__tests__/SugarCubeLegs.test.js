import React from 'react';
import { create } from 'react-test-renderer';

import SugarCubeLegs from '../SugarCubeLegs';

describe('COMPONENT - Characters SugarCat SugarCubeLegs', () => {
  it('renders correctly', () => {
    const component = create(<SugarCubeLegs />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
