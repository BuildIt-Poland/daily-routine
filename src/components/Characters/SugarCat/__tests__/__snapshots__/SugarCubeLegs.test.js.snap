// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Characters SugarCat SugarCubeLegs renders correctly 1`] = `
Array [
  .c0 {
  position: absolute;
  bottom: 2.1rem;
  left: 23rem;
  -webkit-animation: kRJNIP 2.5s ease-in-out -0.3s infinite;
  animation: kRJNIP 2.5s ease-in-out -0.3s infinite;
}

<svg
    className="c0"
    height="22"
    viewBox="0 0 12 22"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6.788 20.234c.569-2.971-.259-5.967-.197-8.793C6.4 9.888 6.24 8.258 5.965 6.7c-.144-.797-.481-1.568-.766-2.142-.15-.311-.248-.678-.419-.968-.154-.261-.378-.574-.523-.82C3.723 2.04 2.71.535 1.629.132.719-.207.14.164 0 .605 1.012 1.606 2.728 3.31 3.197 3.253c-.192.42.408.875.584 1.139.45.688.59 1.522.673 2.324.129 1.237.114 3.525.129 3.416.093.357.191.719.264 1.082a.657.657 0 0 0 .145-.228c.031.616.124 1.04.145 1.542.01.264-.135 1.558-.119 1.604.414 1.03.305 2.79.202 3.866-.067.699-.637 2.148-.388 2.769.109.274 1.661.792 1.811.595.916.15 4.089.714 4.285-.807.286-2.246-3.399-.673-4.14-.321z"
      fill="#222"
    />
  </svg>,
  .c0 {
  position: absolute;
  bottom: 2rem;
  left: 23.8rem;
  -webkit-animation: cpXlhx 2.5s ease-in-out infinite;
  animation: cpXlhx 2.5s ease-in-out infinite;
}

<svg
    className="c0"
    height="23"
    viewBox="0 0 12 23"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6.996 20.792c1.009-3.069.15-6.226.445-9.187-.109-1.532-.305-3.064-.404-4.627-.047-.73-.094-1.704-.27-2.123C6.461 4.129 3.558.913 2.145.373.87-.385-.21.076.035 1.193c1.099.366 3.938 2.626 3.757 2.618a.54.54 0 0 0 .176.341c1.884 1.403 1.506 4.182 1.144 6.324l.078-.31c.083.362.16.725.217 1.092a.634.634 0 0 0 .191-.217c-.062.58.057 1.071.041 1.547a13.68 13.68 0 0 1-.104 1.17c-.005.021-.104.538-.166.45.393 1.051-.057 2.407.062 3.519.093.901.15 1.444-.057 2.489-.171.859-.388.942.238 1.309.44.259 1.133.388 1.61.574.813.326 3.757 1.516 3.778-.274.017-1.789-3.016-1.209-4.004-1.033z"
      fill="#222"
    />
  </svg>,
]
`;
